sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/common/SchemaManager.java

```diff
@@ -164,6 +195,72 @@ public Uni<Schema.CqlTable> getTableAuthorized(
             });
   }
 
+  /**
+   * Get all authorized tables from the bridge.
+   *
+   * <p>Emits a failure in case:
+   *
+   * <ol>
+   *   <li>Not authorized, with {@link UnauthorizedTableException}
+   * </ol>
+   *
+   * @param keyspace Keyspace name
+   * @param missingKeyspace Function of the keyspace in case it's not existing. Usually there to
+   *     provide a failure.
+   * @return Multi of Schema.CqlTable
+   */
+  @WithSpan
+  public Multi<Schema.CqlTable> getTablesAuthorized(
+      String keyspace,
+      Function<String, Uni<? extends Schema.CqlKeyspaceDescribe>> missingKeyspace) {
+    StargateBridge bridge = requestInfo.getStargateBridge();
+
+    // get keyspace
+    return getKeyspaceInternal(bridge, keyspace)
+
+        // if keyspace not found switch to function
+        .onItem()
+        .ifNull()
+        .switchTo(missingKeyspace.apply(keyspace))
+
+        // if it exists go forward
+        .onItem()
+        .ifNotNull()
+        .transformToMulti(
+            keyspaceDescribe -> {
+
+              // create schema reads for all tables
+              List<Schema.CqlTable> tables = keyspaceDescribe.getTablesList();
+              List<Schema.SchemaRead> reads =
+                  tables.stream()
+                      .map(t -> SchemaReads.table(keyspace, t.getName(), sourceApi))
+                      .collect(Collectors.toList());
+
+              Schema.AuthorizeSchemaReadsRequest request =
+                  Schema.AuthorizeSchemaReadsRequest.newBuilder().addAllSchemaReads(reads).build();
+
+              // execute request
+              return bridge
+                  .authorizeSchemaReads(request)
+
+                  // on response filter out
+                  .onItem()
+                  .ifNotNull()
+                  .transformToMulti(
+                      response -> {
+                        List<Schema.CqlTable> authorizedTables = new ArrayList<>(tables);
+                        authorizedTables.removeIf(
+                            table -> {
+                              int index = tables.indexOf(table);
+                              return !Boolean.TRUE.equals(response.getAuthorized(index));
```

Comment:
Apply same as suggested in https://github.com/stargate/stargate/pull/1844#discussion_r881927842

Reply:
Done.